test/tools/yulInterpreter/Interpreter.h

```diff
@@ -136,17 +136,17 @@ class Interpreter: public ASTWalker
 	std::vector<dev::u256> evaluateMulti(Expression const& _expression);
 
 	void openScope() { m_scopes.push_back({}); }
-	/// Unregisters variables.
+	/// Unregisters variables and functions.
 	void closeScope();
 
 	Dialect const& m_dialect;
 	InterpreterState& m_state;
 	/// Values of variables.
 	std::map<YulString, dev::u256> m_variables;
-	/// Meanings of functions.
-	std::map<YulString, FunctionDefinition const*> m_functions;
-	/// Scopes of variables and functions, used to clear them at end of blocks.
-	std::vector<std::set<YulString>> m_scopes;
+	/// Scopes of variables and functions. Used for lookup, clearing at end of blocks
+	/// and passing over the visible functions across function calls.
+	/// The pointer is nullptr if and only if the key is a variable.
+	std::vector<std::map<YulString, FunctionDefinition const*>> m_scopes;
```

Comment:
Wondering if having a separate scope object for variables and functions would make the code more readable. I'm not sure.